### ISSUE TITLE
Investigate reported E&A shapefile upload issue

### ISSUE THREAD

**Issue by Jeanne-le-Roux** [US-GHG-Center/veda-config-ghg#648]
Received this email from Emily at the project office:

Hi Jeanné!

I’m seeing some glitches in the “upload custom areas” utility in the EnA environment. I downloaded Indianapolis city boundaries as a GEOJSON, and they look fine when I do a quick plot myself (blue test.png, included). I get a triangle in EnA! I’ve attached the GEOJSON for reference.

Also, myself and Hannah Zafar both tried uploading some shapefiles recently. We each downloaded a .zip from various sources via Google search (an approach I assume a standard user might take), and then attempted to upload to EnA. I tried one that said “File successfully uploaded” but then didn’t show anything when I added it to EnA, and then with another I got an error saying “Invalid GEOJSON.” Can you guys replicate either of these issues on your side? Attached is the first .zip file I attempted with. 

![Image](https://github.com/user-attachments/assets/1b80a4b8-1b5e-44ba-a735-9cfe087489f4)

![Image](https://github.com/user-attachments/assets/376f5927-da53-408d-976f-46b1fe3b67c3)

Example geojson and shapefile can be found here: https://drive.google.com/drive/folders/1dVVXp309rdC_dYp4hyo_Fs1e-YoNQpB7

**Comment by Jeanne-le-Roux** [comment]
@aboydnw I'm wondering if someone from the VEDA team can look into this issue soon - it was discovered when prepping for the AGU workshop.

**Comment by aboydnw** [comment]
@Jeanne-le-Roux I added it to our bug board and we can look to pull it in an upcoming sprint. I'll try to replicate.

**Comment by aboydnw** [comment]
When I upload the file, I get the same result with some small AOIs and a large triangle. I also get this message in the initial dialog box when uploading:
![image](https://github.com/user-attachments/assets/b33394ae-e6c3-4fac-92d2-19b39e3b3b33)

It seems like we input some logic to simplify a shapefile, likely for performance reasons. I'm guessing that those small areas (not the big triangle) are taking up a lot of the limit of 100 points. Maybe there are also rings in the original file that are being removed. I wonder if there is a simpler shapefile that they could use, as a workaround? 

A fix might be to loosen the limits, but might be worth checking if there is another file that would avoid this simplification.

**Comment by Jeanne-le-Roux** [comment]
I don't know if a simpler shapefile is available, but I can ask - the feedback we're getting is that any shapefiles they're trying to use are not working at all, and the geojsons are being simplified to the point that they're either not useful anymore, or they're not appearing on the map (something is happening to where they're unable to see the shape that they've successfully uploaded). 

 I just googled a shapefile for Alabama. That and the Alabama geojson seemed to work fine. I also grabbed polygon and line shapefiles for Madison County, and Huntsville City limits. The 'line' files don't work at all (after I upload nothing happens). The Madison County_polygon shapefile seems to work well. The polygon for Huntsville City limits looks weird, probably because it had "rings" that were removed. I think this is the same issue they're encountering for the Indianapolis shapefile.  I've added these files to the folder for reference.   

It might just be that the way the parameters are set do not accommodate more detailed type of shapefiles that you might find for a city or similar style boundary. 

If it's failing more often than not, we may want to revisit the current parameters or more clearly communicate the file requirements to the user up front.

@siddharth0248 mentioned having success with some more complex shapefiles in the past. This is totally anecdotal but I wonder if the feature has always been this way or if anything has changed over time.

**Comment by deborahUAH** [comment]
Comment from Emily sent today asked whether this problem will be fixed before AGU?  @aboydnw can you update whether VEDA is handling this issue in FY25.1.4?

**Comment by aboydnw** [comment]
We don't have time to look at this holistically before AGU. One quick fix we might be able to do is increase the max number of points. I think @amarouane-ABDELHAK mentioned there might have been some limits on file size in the past that aren't relevant anymore?

@hanbyul-here do you know where this limit is defined and if it would be simple to increase the limit? Something like 1000 maybe? Municipal city limits seem pretty complex/detailed

Alternatively @deborahUAH , @Jeanne-le-Roux mentioned they might be able to just choose different examples for AGU, with simpler shapefiles that don't exceed the current file limitations.

**Comment by hanbyul-here** [comment]
I think there can be multiple issues intertwined here. 

I do not have access to the folder withe files now, so I would like to comment about overly-simplified polygon issues first. 
I had to dig up some contexts about why we are simplifying the polygons. 

1. URL becomes too long for Cloudfront https://github.com/US-GHG-Center/veda-config-ghg/issues/401 - but it should not affect the current functionality of E&A. So ignoring it for now. 
2. AWS WAF limitation: We found out that AWS WAF has limitation of 16kb (which was bumped up to 64kb later) while debugging Alaska issue before: https://github.com/NASA-IMPACT/veda-ui/issues/923  It is a bit challenging to figure out how much we can simplify the polygons while taking full advantage of AWS WAF from the client side. I increased the limit in this preview: https://deploy-preview-664--ghg-demo.netlify.app/exploration Please  try the preview and let me know if this seems to work for most of cases.

**Comment by hanbyul-here** [comment]
I have an access to the folder now!

1. We do not support any geometry other than polygons. So Line geometry is not supposed to work - but I think we can more gracefully fail. 
2. `ua_shp.zip` file has multiple `shp` files in it, and we do not support this type of shape file.  I am not sure how common of a practice to structure a shape file in this way - if this is a common enough practice, we can make it work, but I think we can put a better instruction for now. 
3. I tested most of the polygon files in the folder (both SHP and GeoJSON formats), and it seems to work. The simplification appears to be within an acceptable range. I tried reducing the level of simplification, but the current limit seems to be the threshold needed to stay within the WAF constraints.(We likely cannot make the polygons any more detailed)
